massa-execution/src/interface_impl.rs

```diff
@@ -127,7 +131,7 @@ impl Interface for InterfaceImpl {
     fn set_data_for(
         &self,
         address: &assembly_simulator::Address,
-        key: &str,
+        key: &assembly_simulator::MassaHash,
```

Comment:
Not your fault but https://github.com/massalabs/massa/pull/2145 is still not merged, and those changes are gonna create conflicts. For eample, `key` is not pre-hashed anymore.

Could you base your work on https://github.com/massalabs/massa/pull/2145 ?

Reply:
I think either PR can rebase and fix conflicts, depending on which one merges first.

Reply:
https://github.com/massalabs/massa/pull/2145 is merged. You should rebase this one on main